### PR TITLE
PR for Issue 13 (adding licensing file information)

### PR DESCRIPTION
**Pull request by aneeshs1729**
There are a total of 3 commits in this PR.
The first commit is for modifications needed in cpp files to add the licensing info
The 2nd one is for modifications needed in .m files to add the licensing info (50% of the matlab files)
The 3rd one is for modifications needed in the remaining .m files.

### CODE REVIEW

**Comment by valeriabarra** [comment]
HI @aneeshs1729 , thank you so much for your contributions! 

Please use language in the PR conversation that will automatically close the associated Issue once the PR is merged, e.g., 

will close #13 

Thank you!

File: src/cpp/divergence.cpp [hunk 1]
```diff
@@ -5,7 +5,21 @@
  *
```
Comment (jbrzensk) [inline src/cpp/divergence.cpp:5]: License information needs to go BEFORE this line.

File: src/cpp/divergence.cpp [hunk 2]
```diff
@@ -5,7 +5,21 @@
...
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * LICENSE file or on the web GNU General Public License 
+ * <https:*www.gnu.org/licenses/> for more details.
+ *
+ **/
```
[the remark this answers is not in the document]
Comment (valeriabarra) [reply src/cpp/divergence.cpp:22]: I think the `**/` was correct here as it is closing the opening on line 1. 
However, as mentioned above, this licensing content should go before line 1.

File: src/cpp/Makefile [hunk 3]
```diff
@@ -1,3 +1,19 @@
+ # 
+ # SPDX-License-Identifier: GPL-3.0-only
+ # 
+ # Copyright 2008-2024 San Diego State University (SDSU) and Contributors
```
Comment (valeriabarra) [inline src/cpp/Makefile:4]: Hi @aneeshs1729 , thank you so much for your contribution! 

We need the following line:
```suggestion
 Copyright 2008-2024 San Diego State University Research Foundation (SDSURF).
```

Please change it in all fines. Thank you!

Comment (aneeshs1729) [reply src/cpp/Makefile:4]: Hello @valeriabarra 
I took your change to heart and changed all the copyright info to all of the files

Comment (jbrzensk) [reply src/cpp/Makefile:4]: Makefile still has the old reference to "contributors"

File: src/cpp/divergence.h [hunk 4]
```diff
@@ -1,11 +1,29 @@
-/**
+/*
```
Comment (jbrzensk) [inline src/cpp/divergence.h:1]: You start with a ```/**``` here, and a ```/*``` for line two, then the SPDX. Other files start with a ```/*``` on line one and two blank lines lines. Pick one, and be consistent across all files. (see gradient.cpp, mixedbc.cpp, utils.cpp )

Comment (aneeshs1729) [reply src/cpp/divergence.h:1]: Hi Jared - 

I have changed the files to be uniform and be consistent

Comment (jbrzensk) [reply src/cpp/divergence.h:1]: Nitpicking here, the top lines are not consistent. Here you have ```/* ``` then the SPDX license identifier.

In other files, you have ```/*``` then a blank line, then the SPDX info.

File: src/cpp/mixedbc.cpp [hunk 5]
```diff
@@ -1,3 +1,29 @@
+/* 
+ *
```
Comment (jbrzensk) [inline src/cpp/mixedbc.cpp:2]: Different spacing

File: src/cpp/utils.cpp [hunk 6]
```diff
@@ -1,12 +1,30 @@
-/**
+/*
+ *
```
Comment (jbrzensk) [inline src/cpp/utils.cpp:2]: two blank lines

File: src/cpp/divergence.h (hunk 4 above)
Comment (jbrzensk) [reply src/cpp/divergence.h:1]: Some files are still different than each other.
divergence.cpp and divergence.h are just two examples.
there is a difference between files starting as 
```
/*
 * 
 * SPDX-License-Identifier: GPL-3.0-only 
```
and
```
/*
 * SPDX-License-Identifier: GPL-3.0-only 
```

**Review by jbrzensk — APPROVED** [review]

**Comment by jbrzensk** [comment]
Merging this, I manually added link to close #13

**Pull request merged**